misk-aws/src/main/kotlin/misk/jobqueue/sqs/FakeAwsSqsModule.kt

```diff
@@ -0,0 +1,108 @@
+package misk.jobqueue.sqs
+
+import com.google.common.util.concurrent.AbstractScheduledService
+import com.google.common.util.concurrent.Service
+import misk.inject.KAbstractModule
+import misk.jobqueue.Job
+import misk.jobqueue.JobConsumer
+import misk.jobqueue.JobHandler
+import misk.jobqueue.JobQueue
+import misk.jobqueue.QueueName
+import misk.tokens.TokenGenerator
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
+
+class FakeAwsSqsModule : KAbstractModule() {
+  override fun configure() {
+    bind<JobQueue>().to<FakeSqs>()
+    bind<JobConsumer>().to<FakeSqs>()
+    multibind<Service>().to<FakeSqs>()
+  }
+}
+
+class FakeJob(
+  override val queueName: QueueName,
+  override val id: String,
+  override val body: String,
+  override val attributes: Map<String, String>
+) : Job {
+  var acknowledged: Boolean = false
+  var deadLettered: Boolean = false
+  override fun acknowledge() {
+    acknowledged = true
+  }
+
+  override fun deadLetter() {
+    deadLettered = true
+  }
+}
+
+/**
+ * Fake for testing and development. Doesn't handle [misk.jobqueue.TransactionalJobQueue]s
+ */
+
+@Singleton
+class FakeSqs @Inject internal constructor(
+  private val tokenGenerator: TokenGenerator
+) : AbstractScheduledService(), JobQueue, JobConsumer {
```

Comment:
Usually with fakes like this, you want to have explicit control over when tasks get dequeued and processed so that you can write tests that are deterministic and avoid race conditions.  There's a class `ExplicitReleaseBlockingQueue` specifically for this purpose; I'd recommend having the `FakeSqs` use this blocking queue and expose a method to explicit release 1 / n / all of the jobs currently on the queue. That way tests have full control over when jobs will run.